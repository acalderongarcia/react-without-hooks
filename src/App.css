*{
  font-family: 'Roboto', sans-serif;
}

.App {
  text-align: center;
}

.firstSection{
  background-color:transparent !important;
}

.secondSection{
  background-color:rgb(243, 242, 242) !important;
}

.logo{
  height: 325px;
}

.downloadApple{
  height: 50px;
}

.blue-text{
  color: rgb(28, 69, 255);
}

.btn{
  border-radius: 0!important;
}

.btn-blue{
  background-color: #1976d2!important;
  color: #fff!important;
}
