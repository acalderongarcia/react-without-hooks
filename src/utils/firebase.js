import firebase from 'firebase/app';

export const config = {
    apiKey:"AIzaSyDOHJKYYhRrOs6lfqoQNB1_2upXovuQIqY",
    authDomain:"navigation-auth-1556519442679.firebaseapp.com"
};

firebase.initializeApp(config);
export default firebase;
