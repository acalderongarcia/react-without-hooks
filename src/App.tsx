import React from 'reactn';
import './App.css';
import 'bootstrap/dist/css/bootstrap.css';
import Header from "./components/Header";
import PropTypes from "prop-types";
import t from "./utils/localization/Localization";

class App extends React.Component{

    static propTypes = {
        children: PropTypes.object.isRequired
    };

    render() {
        t.setLanguage(this.global.currentLanguage)
        const children = this.props;
        return (
            <div>
                <Header/>
                {children.children}

                <body> 
                sssFFFFFFFFFFFFFFFFFF
                
                </body>
            </div>
        );
    }
}

export default App;
