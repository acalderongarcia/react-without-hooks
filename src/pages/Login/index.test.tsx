import * as React from 'reactn';
import renderer from 'react-test-renderer';
import '../../App.css';
import 'bootstrap/dist/css/bootstrap.css';
import t from "../../utils/localization/Localization"
import Button from '@material-ui/core/Button';
import Form from 'react-bootstrap/Form';
import 'firebase/auth';
import StyledFirebaseAuth from "react-firebaseui/StyledFirebaseAuth";
import firebase from '../../utils/firebase';
import Validate from '../../utils/validator/Validator';
import firebaseui from 'firebaseui';

const deleteAccount = () => {
    // @ts-ignore
    firebase.auth().currentUser.delete().catch((error) => {
        if (error.code === 'auth/requires-recent-login') {
            firebase.auth().signOut().then(() => {
                setTimeout(() => {
                    alert('Please sign in again to delete your account.');
                }, 1);
            });
        }
    });
}

describe('Testing src/pages/Login/index.tsx', () => {

    test('Testing jumbotron firstSection text-center with isSignedIn = true', () => {
        // Before
        let isSignedIn = true;
        let uiConfig: firebaseui.auth.Config = {
            signInFlow: "popup",
            signInOptions: [
                firebase.auth.GoogleAuthProvider.PROVIDER_ID,
                firebase.auth.FacebookAuthProvider.PROVIDER_ID
            ]
        }

        // CAMBIAR:        
        // <Form.Group controlId="formBasicEmail">
        // <Form.Group controlId="formBasicPassword">
        // POR:
        // <Form.Group>
        // <Form.Group>

        const wrapper = renderer.create(
            <div>
                <div className="jumbotron firstSection text-center">
                    <h1 className="card-title h1 my-5">
                        {t.login}
                    </h1>
                    <Form>
                        <Form.Group>
                            <Form.Label className="blue-text my-3 font-weight-bold">{t.email}</Form.Label>
                            <Form.Control id="email" type="email" placeholder={t.placeHolderEmail} required />
                        </Form.Group>
                        <Form.Group>
                            <Form.Label className="blue-text my-3 font-weight-bold">{t.password}</Form.Label>
                            <Form.Control id="password" type="password" placeholder={t.placeHolderPassword} required />
                        </Form.Group>
                        <Button onClick={onClickLink} type="button" className="pl-4 pr-4 btn mr-3 btn-blue waves-effect">
                            {t.login}
                        </Button>
                        <p></p>
                        {isSignedIn !== undefined && !isSignedIn &&
                            <StyledFirebaseAuth uiConfig={uiConfig} firebaseAuth={firebase.auth()} />
                        }
                        {isSignedIn &&
                            <div>
                                <Button type="button" onClick={() => firebase.auth().signOut()}>Sign-out</Button>
                                <Button type="button" onClick={deleteAccount}>Delete account</Button>
                            </div>
                        }
                    </Form>
                </div>
            </div>
        );
        // Then
        let tree = wrapper.toJSON();
        // After
        expect(tree).toMatchSnapshot();
    });

    test('Testing jumbotron firstSection text-center with isSignedIn = false', () => {
        // Before
        let isSignedIn = false;
        let uiConfig: firebaseui.auth.Config = {
            signInFlow: "popup",
            signInOptions: [
                firebase.auth.GoogleAuthProvider.PROVIDER_ID,
                firebase.auth.FacebookAuthProvider.PROVIDER_ID
            ]
        }
        const wrapper = renderer.create(
            <div>
                <div className="jumbotron firstSection text-center">
                    <h1 className="card-title h1 my-5">
                        {t.login}
                    </h1>
                    <Form>
                        <Form.Group>
                            <Form.Label className="blue-text my-3 font-weight-bold">{t.email}</Form.Label>
                            <Form.Control id="email" type="email" placeholder={t.placeHolderEmail} required />
                        </Form.Group>
                        <Form.Group>
                            <Form.Label className="blue-text my-3 font-weight-bold">{t.password}</Form.Label>
                            <Form.Control id="password" type="password" placeholder={t.placeHolderPassword} required />
                        </Form.Group>
                        <Button onClick={onClickLink} type="button" className="pl-4 pr-4 btn mr-3 btn-blue waves-effect">
                            {t.login}
                        </Button>
                        <p></p>
                        {isSignedIn !== undefined && !isSignedIn &&
                            <StyledFirebaseAuth uiConfig={uiConfig} firebaseAuth={firebase.auth()} />
                        }
                        {isSignedIn &&
                            <div>
                                <Button type="button" onClick={() => firebase.auth().signOut()}>Sign-out</Button>
                                <Button type="button" onClick={deleteAccount}>Delete account</Button>
                            </div>
                        }
                    </Form>
                </div>
            </div>
        );
        // Then
        let tree = wrapper.toJSON();
        // After
        expect(tree).toMatchSnapshot();
    });

});

function onClickLink() {
    Validate.emailResult = Validate.schemaEmail.validate((document.getElementById('email') as HTMLInputElement).value),
        Validate.passResult = Validate.schemaPass.validate((document.getElementById('password') as HTMLInputElement).value)

    if (Validate.emailResult.error) {
        (document.getElementById('email') as HTMLInputElement).value = '';
        alert(t.loginEmailError);
    }
    if (Validate.passResult.error) {
        (document.getElementById('password') as HTMLInputElement).value = '';
        alert(t.loginPassError);
    }
}

