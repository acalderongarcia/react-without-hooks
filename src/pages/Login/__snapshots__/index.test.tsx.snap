// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing src/pages/Login/index.tsx Testing jumbotron firstSection text-center with isSignedIn = false 1`] = `
<div>
  <div
    className="jumbotron firstSection text-center"
  >
    <h1
      className="card-title h1 my-5"
    >
      Login
    </h1>
    <form
      className=""
    >
      <div
        className="form-group"
      >
        <label
          className="blue-text my-3 font-weight-bold form-label"
        >
          Email
        </label>
        <input
          className="form-control"
          id="email"
          placeholder="Enter email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="blue-text my-3 font-weight-bold form-label"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 pl-4 pr-4 btn mr-3 btn-blue waves-effect"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-2"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-30"
        />
      </button>
      <p />
      <div
        id="firebaseui_container"
      />
    </form>
  </div>
</div>
`;

exports[`Testing src/pages/Login/index.tsx Testing jumbotron firstSection text-center with isSignedIn = true 1`] = `
<div>
  <div
    className="jumbotron firstSection text-center"
  >
    <h1
      className="card-title h1 my-5"
    >
      Login
    </h1>
    <form
      className=""
    >
      <div
        className="form-group"
      >
        <label
          className="blue-text my-3 font-weight-bold form-label"
        >
          Email
        </label>
        <input
          className="form-control"
          id="email"
          placeholder="Enter email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="blue-text my-3 font-weight-bold form-label"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6 pl-4 pr-4 btn mr-3 btn-blue waves-effect"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-2"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-30"
        />
      </button>
      <p />
      <div>
        <button
          className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-2"
          >
            Sign-out
          </span>
          <span
            className="MuiTouchRipple-root-30"
          />
        </button>
        <button
          className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-2"
          >
            Delete account
          </span>
          <span
            className="MuiTouchRipple-root-30"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
